### DISCUSSION TITLE
Access property backing field from inside property

### DISCUSSION THREAD

**Discussion by DJDoena** [dotnet/csharplang#7742] [Language Ideas] [answered]
if you have a standard auto-implemented Property
```
private int MyProp { get; set; }
```
C# will generate a backing field in the compiled code that is not accessible (outside reflection) from the rest of the class.

However if you need to do some value checks like this:

```
private int MyField;
private int MyProp
{
  get => MyField; 
  set
  {
    if (!IsValid(value))
    {
      throw new Exception();
    }
    MyField = value;
  }
 }
```
you need an explicit field that is now accessible from the rest of the code. It would be great to have the backing field only accessible for the property itself, like this:
```
private int MyProp
{
  get => _MyProp; //implicit name based on property
  set
  {
    if (!IsValid(value))
    {
      throw new Exception();
    }
    _MyProp = value;
  }
 }
```

or like this

```
private int MyProp
{
  int MyField; //explicit declaration
  get => MyField; 
  set
  {
    if (!IsValid(value))
    {
      throw new Exception();
    }
    MyField = value;
  }
 }
```
in both proposal cases the backing field should not be accessible by regular code outside the property itself

**Comment by AndersKehlet** [comment, marked as answer]
https://github.com/dotnet/csharplang/blob/4dba3f29032a8164fa243a39b56e3a426a7995fc/proposals/semi-auto-properties.md

They're working on it.

**Reply by DavidArno** [reply]
> They're working on it.

It's worth clarifying that the team do not appear to be actively "working on it" any more as it's blocked by unresolved issues around breaking changes (#7033).

**Reply by AndersKehlet** [reply]
Unresolved issues that they are actively working on (Mads has marked breaking change warnings as ASAP for 2024).
They _could_ implement the field access feature with a conditional keyword today, but they would prefer a "proper" implementation.
They also showed field access at Build as something they very much want to do, so overall I think it's reasonable to say it's being worked on (i.e. there's a design and we're very likely to get this feature soon™️).

**Reply by CyrusNajmabadi** [reply]
> It's worth clarifying that the team do not appear to be actively "working on it" any more as it's blocked by unresolved issues around breaking changes

This is not blocked. We are working on this for c#13. We just had a meeting yesterday continuing things here :-)

**Reply by CyrusNajmabadi** [reply]
> They could implement the field access feature 

Correct. If we have a breaking change system ready, we will use that. If we don't, we will ship `field` in a non-breaking fashion, and then make any breaks later if and when that system becomes available. :-)
